### PR TITLE
Add Asset::removeMetadata() method

### PR DESCRIPTION
**Pull request by blankse**
## Changes in this pull request  
See https://github.com/pimcore/pimcore/discussions/10311

### CODE REVIEW

File: models/Asset.php [hunk 1]
```diff
@@ -1676,6 +1676,34 @@ public function addMetadata($name, $type, $data = null, $language = null)
...
+    {
+        if ($name) {
+            $tmp = [];
+            $name = str_replace('~', '---', $name);
+            if (!is_array($this->metadata)) {
+                $this->metadata = [];
+            }
+
+            foreach ($this->metadata as $item) {
+                if ($item['name'] != $name || $language != $item['language']) {
```
Comment (weisswurstkanone) [inline models/Asset.php]: Is there  a way to delete all items for the given name regardless of the language ?

Comment (blankse) [reply models/Asset.php]: I add language = '*' option

File: models/Asset.php [hunk 2]
```diff
@@ -1676,6 +1676,35 @@ public function addMetadata($name, $type, $data = null, $language = null)
...
+    {
+        if ($name) {
+            $tmp = [];
+            $name = str_replace('~', '---', $name);
+            if (!is_array($this->metadata)) {
+                $this->metadata = [];
+            }
+
+            foreach ($this->metadata as $item) {
+                if ($item['name'] === $name && ($language == $item['language'] || $language === '*')) {
```
Comment (weisswurstkanone) [inline models/Asset.php:1695]: I like the idea. I think the docs should be enriched then accordingly 👍

Comment (blankse) [reply models/Asset.php:1695]: I add it to the docs

**Comment by weisswurstkanone** [comment]
Thanks a lot!

**Pull request merged**